compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -878,36 +1030,48 @@ abstract class GenJSCode extends plugins.PluginComponent
 
       val bodyIsStat = resultIRType == jstpe.NoType
 
-      val body = tree match {
+      def genBody() = tree match {
         case Block(
             (thisDef @ ValDef(_, nme.THIS, _, initialThis)) :: otherStats,
             rhs) =>
           // This method has tail jumps
-          withScopedVars(
-            (initialThis match {
-              case This(_)  =>
-                Seq(methodTailJumpThisSym := thisDef.symbol,
-                    fakeTailJumpParamRepl := (NoSymbol, NoSymbol))
-              case Ident(_) =>
-                Seq(methodTailJumpThisSym := NoSymbol,
-                    fakeTailJumpParamRepl := (thisDef.symbol, initialThis.symbol))
-            }): _*
-          ) {
-            val innerBody = js.Block(otherStats.map(genStat) :+ (
+
+          // To be called from within withScopedVars
+          def genInnerBody() = {
+            js.Block(otherStats.map(genStat) :+ (
                 if (bodyIsStat) genStat(rhs)
                 else            genExpr(rhs)))
+          }
```

Comment:
Consider putting the `withScopedVars` in here. It will improve two things:

- The comment is obsolete and it's explicit from the code
- Less code below